src/main/webapp/app/admin/legal/legal-document-update.component.ts

```diff
@@ -0,0 +1,138 @@
+        } else {
+            this.languageHelper.updateTitle('artemisApp.legal.privacyStatement.updatePrivacyStatement');
+            this.titleKey = 'artemisApp.legal.privacyStatement.updatePrivacyStatement';
+        }
+        this.legalDocument = new LegalDocument(this.legalDocumentType, this.defaultLanguage);
+        this.getLegalDocumentForUpdate(this.legalDocumentType, this.defaultLanguage).subscribe((document) => {
+            this.legalDocument = document;
+        });
+    }
+
+    private getLegalDocumentForUpdate(type: LegalDocumentType, language: LegalDocumentLanguage): Observable<LegalDocument> {
+        if (type === LegalDocumentType.PRIVACY_STATEMENT) {
+            return this.legalDocumentService.getPrivacyStatementForUpdate(language);
+        } else {
+            return this.legalDocumentService.getImprintForUpdate(language);
+        }
+    }
+
+    updateLegalDocument() {
+        this.isSaving = true;
+        this.legalDocument.text = this.markdownEditor.markdown!;
+        if (this.legalDocumentType === LegalDocumentType.PRIVACY_STATEMENT) {
+            this.legalDocumentService.updatePrivacyStatement(this.legalDocument).subscribe((statement) => {
+                this.legalDocument = statement;
+                this.unsavedChanges = false;
+                this.isSaving = false;
+            });
+        } else {
+            this.legalDocumentService.updateImprint(this.legalDocument).subscribe((statement) => {
+                this.legalDocument = statement;
+                this.unsavedChanges = false;
+                this.isSaving = false;
+            });
+        }
+    }
+
+    checkUnsavedChanges(content: string) {
+        this.unsavedChanges = content !== this.legalDocument.text;
+    }
+
+    onLanguageChange(legalDocumentLanguage: any) {
+        if (this.unsavedChanges) {
+            this.showWarning(legalDocumentLanguage);
+        } else {
+            this.markdownEditor.markdown = '';
+            this.currentLanguage = legalDocumentLanguage;
+            this.getLegalDocumentForUpdate(this.legalDocumentType, legalDocumentLanguage).subscribe((document) => (this.legalDocument = document));
+        }
+    }
+
+    showWarning(legalDocumentLanguage: any) {
+        this.unsavedChangesWarning = this.modalService.open(UnsavedChangesWarningComponent, { size: 'lg', backdrop: 'static' });
+        if (this.legalDocumentType === LegalDocumentType.PRIVACY_STATEMENT) {
+            this.unsavedChangesWarning.componentInstance.textMessage = 'artemisApp.legal.privacyStatement.unsavedChangesWarning';
+        } else {
+            this.unsavedChangesWarning.componentInstance.textMessage = 'artemisApp.legal.imprint.unsavedChangesWarning';
+        }
+
+        this.unsavedChangesWarning.result.then(
+            () => {
+                this.unsavedChanges = false;
+                this.onLanguageChange(legalDocumentLanguage);
+            },
+            () => {},
+        );
```

Comment:
```suggestion
            }
        ).catch(() => {
            // ignore
        });
```

I think this would be more idiomatic